Classes/common/touchdb/ChangeTracker/TDURLConnectionChangeTracker.m

```diff
@@ -0,0 +1,298 @@
+    self.request = [[NSMutableURLRequest alloc] initWithURL:url];
+    self.request.cachePolicy = NSURLRequestReloadIgnoringLocalCacheData;
+    self.request.HTTPMethod = @"GET";
+    
+    // Add headers from my .requestHeaders property:
+    for(NSString *key in self.requestHeaders) {
+        [self.request setValue:self.requestHeaders[key] forHTTPHeaderField:key];
+    }
+    
+    // Add cookie headers from the NSHTTPCookieStorage:
+    NSArray* cookies = [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookiesForURL:url];
+    NSDictionary* cookieHeaders = [NSHTTPCookie requestHeaderFieldsWithCookies:cookies];
+    for (NSString* headerName in cookieHeaders) {
+        [self.request setValue:cookieHeaders[headerName] forHTTPHeaderField:headerName];
+    }
+
+    if (self.authorizer) {
+        NSString* authHeader = [self.authorizer authorizeURLRequest:self.request forRealm:nil];
+        if (authHeader) {
+            [self.request setValue: authHeader forHTTPHeaderField:@"Authorization"];
+        }
+    }
+    
+    self.connection = [[NSURLConnection alloc] initWithRequest:self.request
+                                                      delegate:self
+                                              startImmediately:NO];
+
+    if (!self.connection) {
+        CDTLogError(CDTREPLICATION_LOG_CONTEXT, @"%@: GET %@ failed to create NSURLConnection",
+                   self, @"cleaned url.resourceSpecifier");
+        return NO;
+    }
+    [self.connection scheduleInRunLoop:[NSRunLoop currentRunLoop] forMode:NSRunLoopCommonModes];
+    [self.connection start];
+    
+    self.inputBuffer = [NSMutableData dataWithCapacity:0];
+        
+    self.startTime = [NSDate date];
+    CDTLogInfo(CDTREPLICATION_LOG_CONTEXT, @"%@: Started... <%@>", [self class], self.changesFeedURL);
+    CDTLogVerbose(CDTREPLICATION_LOG_CONTEXT, @"%@: GET %@", self, url.resourceSpecifier);
+
+    return YES;
+}
+
+- (void)clearConnection
+{
+    [self.connection cancel];
+    [self.connection unscheduleFromRunLoop:[NSRunLoop currentRunLoop] forMode:NSRunLoopCommonModes];
+    self.connection = nil;
+    self.inputBuffer = nil;
+}
+
+- (void)stop
+{
+    [NSObject cancelPreviousPerformRequestsWithTarget:self
+                                             selector:@selector(start)
+                                               object:nil];  // cancel pending retries
+    if (self.connection) {
+        CDTLogInfo(CDTREPLICATION_LOG_CONTEXT, @"%@: stop", [self class]);
+        [self clearConnection];
+    }
+    [super stop];
+}
+
+- (void)retryOrError:(NSError*)error
+{
+    CDTLogInfo(CDTREPLICATION_LOG_CONTEXT, @"%@: retryOrError: %@", [self class], error);
+    if (++_retryCount <= kMaxRetries && TDMayBeTransientError(error)) {
+        self.totalRetries++;
+        [self clearConnection];
+        NSTimeInterval retryDelay = kInitialRetryDelay * (1 << (_retryCount - 1));
+        [self performSelector:@selector(start) withObject:nil afterDelay:retryDelay];
```

Comment:
Is `-performSelector:...` going to be be called on the replicator's thread? I'm assuming it will because from what I can see `-retryOrError` is called from the NSURLConnection callbacks which should be called in the replicator's thread due to the run queue for that thread being used. Can you confirm that?